### PR TITLE
Fix incorrect URL_REALMS in keycloak.py 

### PR DESCRIPTION
**Pull request by Gaetan2907**
##### SUMMARY
The URL_REALMS in `keycloak.py` should be `"{url}/admin/realms"` instead of `"{url}/realms"` 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
`keycloak.py` 

##### ADDITIONAL INFORMATION
After testing I got an error 404 with the `"{url}/realms"`. 
Here is where I found the right URL to use : 
https://www.lisenet.com/2020/create-a-keycloak-realm-using-admin-rest-api/

### CODE REVIEW

File: plugins/module_utils/identity/keycloak/keycloak.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 from ansible.module_utils.six.moves.urllib.error import HTTPError
 from ansible.module_utils._text import to_native
 
-URL_REALMS = "{url}/realms"
+URL_REALMS = "{url}/admin/realms"
```
Comment (aminvakil) [inline plugins/module_utils/identity/keycloak/keycloak.py:40]:
```suggestion
URL_REALMS = "{url}/auth/admin/realms"
```

**Review by aminvakil — COMMENTED** [review]
Admin url should be `/auth/admin/realms` from your link and from https://www.keycloak.org/docs/latest/server_admin/index.html#admin-console.

File: plugins/module_utils/identity/keycloak/keycloak.py (hunk 1 above)
Comment (Gaetan2907) [reply plugins/module_utils/identity/keycloak/keycloak.py:40]: Yes you're right, this is the right URL to use but according to the example given at the line 504 of `keycloak.py` we should use https://auth.example.com/auth to call the module. So I think either the example must be change either URL_REALMS as I suggested. With your suggestion the URL used by the module will become  https://auth.example.com/auth/auth/admin/realms and this is not what we want. 
I hope my explanation was clear enough

Comment (felixfontein) [reply plugins/module_utils/identity/keycloak/keycloak.py:40]: I think you mean `keycloak_realm.py` and not `keycloak.py`? In any case, in the examples of all keycloak modules, `/auth` is part of the URL.

**Comment by felixfontein** [comment]
This was added in #2155 by @kris2kris, which was only merged three days ago. @kris2kris can you take a look at this PR?

**Comment by kris2kris** [comment]
Effectively, the url should be `{url}/admin/realms`. You can compare the urls for clients or for groups in the `keycloak.py`, every URLs must have the same "pattern".
It's probably a mistake when a recopied it in the file 😞

**Comment by kris2kris** [comment]
The URL should be `{url}/admin/realms` to be aligned with other URLs from the file (clients or groups), `auth` is not needed.
I forgot the `admin` part when I recopied URLs in file 😞

**Comment by felixfontein** [comment]
@kris2kris so `URL_REALM` and `URL_TOKEN` are ok (without `/admin/`), but the change to `URL_REALMS` in this PR is needed?

**Comment by kris2kris** [comment]
> URL_REALM



> @kris2kris so `URL_REALM` and `URL_TOKEN` are ok (without `/admin/`), but the change to `URL_REALMS` in this PR is needed?

The two URL added in #2155 (`URL_REALM` and `URL_REALMS`) must be changed, all others are ok and was not changed for a long time.

File: plugins/module_utils/identity/keycloak/keycloak.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
...
 URL_REALM = "{url}/realms/{realm}"
```
Comment (felixfontein) [inline plugins/module_utils/identity/keycloak/keycloak.py:41]: So according to @kris2kris this also needs to be changed:
```suggestion
URL_REALM = "{url}/admin/realms/{realm}"
```

File: plugins/module_utils/identity/keycloak/keycloak.py (hunk 1 above)
Comment (Gaetan2907) [reply plugins/module_utils/identity/keycloak/keycloak.py:40]: Yes sorry I meant `keycloak_realm.py` 
That's why I think using this line : `URL_REALMS = "{url}/admin/realms` would make the usage of the realm similar as the other

**Comment by felixfontein** [comment]
If everyone's happy, I'll merge this.

**Review by aminvakil — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@Gaetan2907 thanks for fixing this!
@aminvakil @kris2kris thanks for reviewing!

**Pull request merged**